engines/testbed/video.cpp

```diff
@@ -153,17 +176,21 @@ Common::Error Videotests::videoTest(Common::SeekableReadStream *stream, const Co
 						mouse.y >= y && mouse.y < y + mh) {
 					switch (event.type) {
 					case Common::EVENT_LBUTTONDOWN:
-						((Video::QuickTimeDecoder *)video)->handleMouseButton(true, event.mouse.x - x, event.mouse.y - y);
+						if (qtVideo)
+							qtVideo->handleMouseButton(true, event.mouse.x - x, event.mouse.y - y);
 						break;
 					case Common::EVENT_LBUTTONUP:
-						((Video::QuickTimeDecoder *)video)->handleMouseButton(false, event.mouse.x - x, event.mouse.y - y);
+						if (qtVideo)
+							qtVideo->handleMouseButton(false, event.mouse.x - x, event.mouse.y - y);
 						break;
 					case Common::EVENT_MOUSEMOVE:
-						((Video::QuickTimeDecoder *)video)->handleMouseMove(event.mouse.x - x, event.mouse.y - y);
+						if (qtVideo)
+							qtVideo->handleMouseMove(event.mouse.x - x, event.mouse.y - y);
 						break;
 					case Common::EVENT_KEYUP:
 					case Common::EVENT_KEYDOWN:
-						((Video::QuickTimeDecoder *)video)->handleKey(event.kbd, event.type == Common::EVENT_KEYDOWN);
+						if (qtVideo)
+							qtVideo->handleKey(event.kbd, event.type == Common::EVENT_KEYDOWN);
 						break;
 					default:
 						break;
```

Comment:
Since keyboard events are handled pretty much only by QT videos, these QT video checks can be moved into a single check in the enclosing if statement, i.e. something like:

```
				if (Common::isMouseEvent(event))
					mouse = event.mouse;
				if (qtVideo && mouse.x >= x && mouse.x < x + mw &&
						mouse.y >= y && mouse.y < y + mh) {
```

Reply:
Update: Committed proposed fix